src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidations.java

```diff
@@ -51,13 +53,14 @@ private CallbackValidations() {
 
     @NotNull
     static Validation<String, String> hasCaseReference(CaseDetails theCase) {
-        Object caseReference = null;
-        return theCase != null
-            && theCase.getData() != null
-            && (caseReference = theCase.getData().get(ATTACH_TO_CASE_REFERENCE)) != null
-            && (caseReference instanceof String)
-            ? valid((String) caseReference)
-            : internalError("no case reference found: %s", String.valueOf(caseReference));
+        return caseRefValidator.validate(theCase);
+    }
+
+    private static Optional<String> validRef(Object reference) {
```

Comment:
Given the `caseRefValidator` presence, is this method being used?

Reply:
sonar picked this up to 
```
src/.../orchestrator/services/ccd/CallbackValidations.java
Remove this unused private "validRef" method.
yesterday
L59
```